File: frontend/src/types/promo-video.ts
export interface PromoVideo {
  id: string;
  title: string;
  description: string | null;
  videoUrl: string;
  posterUrl: string | null;
  displayOrder: number;
  active: boolean;
  updatedAt: string;
}

export interface UpsertPromoVideoPayload {
  title: string;
  description?: string;
  videoUrl: string;
  posterUrl?: string | null;
  displayOrder?: number;
  active?: boolean;
}

File: src/main/java/com/gamyacouture/site/domain/PromoVideo.java
package com.gamyacouture.site.domain;

import com.gamyacouture.shared.domain.BaseAuditableEntity;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.UUID;

@Entity
@Table(name = "promo_videos")
@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class PromoVideo extends BaseAuditableEntity {

    @Id
    private UUID id;

    @Column(nullable = false, length = 200)
    private String title;

    @Column(columnDefinition = "TEXT")
    private String description;

    @Column(name = "video_url", nullable = false, length = 500)
    private String videoUrl;

    @Column(name = "poster_url", length = 500)
    private String posterUrl;

    @Column(name = "display_order", nullable = false)
    @Builder.Default
    private int displayOrder = 0;

    @Column(nullable = false)
    @Builder.Default
    private boolean active = true;
}
